### PR TITLE
use different ssl dir on AIO

### PR DESCRIPTION
**Pull request by mmoll**
see https://docs.puppetlabs.com/puppet/4.3/reference/whered_it_go.html#ssldir-is-inside-confdir

### CODE REVIEW

**Comment by mmoll** [comment]
looking for comments on that one...

The `puppetssldir` setting is used to derive other paths in the Foreman settings, the symptom I saw was that Foreman could not load the certificate it uses to communicate with the proxy. I have no good ideas how to handle that...

File: manifests/params.pp [hunk 1]
```diff
@@ -215,19 +215,32 @@
...
+  if versioncmp($::puppetversion, '4.0') < 0 {
+    $aio_package = false
+  } elsif $::osfamily == 'Windows' or $::rubysitedir =~ /\/opt\/puppetlabs\/puppet/ {
+    $aio_package = true
+  } else {
+    $aio_package = false
+  }
+
+  $puppet_ssldir = $aio_package ? {
+    true    => '/etc/puppetlabs/puppet/ssl',
```
Comment (ekohl) [inline manifests/params.pp]: What happens on Windows here?

Comment (mmoll) [inline manifests/params.pp]: Windows is ruled out by running into line 212 anyway, I just copypasted line 218-224 from puppet-puppet.

Comment (ekohl) [inline manifests/params.pp]: You're right, we don't fully support that anyway. I was thinking of settings.yaml

**Comment by mmoll** [comment]
yeah, I guess it would work (and I would also apply the usual logic with pulling `$puppet_ssldir` into init.pp to make it overridable), and it does not feel correct with current master anyway, as the Foreman settings will also get not updated when one overrides the respective parameters here...

**Comment by mmoll** [comment]
updated and added a test

**Comment by mmoll** [comment]
@brandonweeks BTW, if you want to test... :point_up:

**Comment by mmoll** [comment]
rebased.

**Comment by ekohl** [comment]
Untested, but in general :+1:. We need a similar fix in puppet-foreman_proxy.

**Comment by mmoll** [comment]
-proxy worked for me as-is, because `$::puppet::params::ssldir` is used there, but maybe that only worked for me because I did an all-in-one setup.

**Comment by brandonweeks** [comment]
Whoops, I thought I commented on this yesterday. Tested :+1: in production. :)

**Comment by domcleal** [comment]
Merged, thanks @mmoll!

**Pull request merged**